docs/cookbook/building-navigation.md

```diff
@@ -0,0 +1,292 @@
+
+![Screenshot of the Apostrophe demo page structure in the page manager UI](/images/recipes/demo-page-tree.png)
+
+Apostrophe templates have access to this page tree using `data.home._children`, an array of top-level page data objects. Templates also have access to the children of the page a visitor is on, but since we're focused on the main site navigation, we want to use the home page's children. Home page children do not change as visitors move around the website.
+
+As explained on the [pages guide](/guide/pages.md#connecting-pages-with-page-tree-navigation), we can loop over `data.home._children` and print the pages' URLs and titles in a list. Once you add some classes and CSS, this can work great as site navigation. And since it is based on the page tree structure **it will automatically stay up to date as editors update page content**.
+
+<AposCodeBlock>
+  ```django
+  <header>
+    <nav>
+      <ul>
+        {# 👇 Referencing `data.home._children` and looping over them. #}
+        {% for page in data.home._children %}
+          <li>
+            <a href="{{ page._url }}">{{ page.title }}</a>
+          </li>
+        {% endfor %}
+      </ul>
+    </nav>
+  </header>
+  ```
+  <template v-slot:caption>
+    views/layout.html
+  </template>
+</AposCodeBlock>
+
+### Change page tree data in templates
+
+The example above is a very simple site navigation. It includes only the top-level pages and the basic data available for each page. In some cases you may want different data from the page tree for your navigation. The direct way to do this is to alter the **query builders** that the core page module uses to get that data.
+
+The *default* query builder options for the page module look like this:
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    options: {
+      builders: {
+        children: true,
+        ancestors: {
+          children: true
+        }
+      }
+    }
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+For example, **to include two levels of pages in the nav** we would add to the `builders` option that ancestors (such as the home page) should include two levels of child pages (`depth: 2`).
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    options: {
+      builders: {
+        children: true,
+        // `ancestors` includes the home page, so we adjust things there.
+        ancestors: {
+          children: {
+            depth: 2
+          }
+        }
+      }
+    }
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+By default the page data object includes all properties from the database document. We can **limit that returned data** (for a minor speed improvement and clearer logging) by adding a projection to the query builder. *Make sure to include the `path` in this projection to avoid crashing.*
```

Comment:
Ouch! But true. Want to open a ticket to automatically include path in projections of pages or something?

Reply:
I'll do that.